### PR TITLE
DeepScan fixes

### PR DESCRIPTION
**Pull request by Arkatos1**
## About The Pull Request

This PR fixes a majority of the current high to medium severity issues detected by the DeepScan application, hopefully improving code quality a little bit. Most of the fixes should be quite straightforward - insufficient or unnecessary null checks (like in the required_experiments case where its assigned as an empty list if its null in the first place), or checking a condition that is constant.

I have tested ingame all fixes that were possible to reasonably test, and found no problems with them.

View fixed issues here:
https://deepscan.io/dashboard/#view=project&tid=12157&pid=15129&bid=298738&prid=&subview=issues&b=master&impact=%5B%22High%22%2C%22Medium%22%5D&page=1

## Why It's Good For The Game

Cleaner code.

### CODE REVIEW

File: tgui/packages/tgui-polyfill/ie8.js [hunk 1]
```diff
@@ -458,7 +458,7 @@
         ) : (
           (parentNode = self.parentNode) /* && live(self) */ ?
             parentNode.dispatchEvent(e) :
-            true
+            null
```
Comment (stylemistake) [inline tgui/packages/tgui-polyfill/ie8.js:461]: You should revert this bit because it's vendored code.

Comment (Arkatos1) [reply tgui/packages/tgui-polyfill/ie8.js]: Understood, was not really sure about this one.

File: tools/build/juke/index.js [hunk 2]
```diff
@@ -3552,12 +3552,12 @@ var require_sync = __commonJS({
...
     };
     GlobSync.prototype._processSimple = function(prefix, index) {
-      var exists = this._stat(prefix);
+      var exists = prefix ? this._stat(prefix) : false;
       if (!this.matches[index])
         this.matches[index] = Object.create(null);
       if (!exists)
         return;
-      if (prefix && isAbsolute(prefix) && !this.nomount) {
+      if (isAbsolute(prefix) && !this.nomount) {
```
Comment (stylemistake) [inline tools/build/juke/index.js:3560]: Same, vendored code, leave it alone.

**Comment by stylemistake** [comment]
Looks good, commented on vendored code.

File: tools/build/juke/index.js (hunk 2 above)
Comment (Arkatos1) [reply tools/build/juke/index.js]: Reverted.

**Review by Cyberboss — APPROVED** [review]

**Pull request merged**